src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,444 @@
+     * @param {Component} toggleWidget - The `gpii.qssWidget.mouseWidgetToggle` instance.
+     * @param {fluid.event} event - onQssWidgetSettingAltered event
+     */
+    gpii.qssWidget.mouseWidgetToggle.toggleModel = function (that, toggleWidget, event) {
+        if (toggleWidget.model.setting.schema.mapOff || toggleWidget.model.setting.schema.mapOn) {
+            if (that.model.enabled) {
+                toggleWidget.model.setting.value = toggleWidget.model.setting.schema.mapOff;
+            } else {
+                toggleWidget.model.setting.value = toggleWidget.model.setting.schema.mapOn;
+            }
+        } else {
+            toggleWidget.model.setting.value = !that.model.enabled;
+        }
+
+        event.fire(toggleWidget.model.setting);
+        that.applier.change("enabled", !that.model.enabled, null, "fromWidget");
+    };
+
+    /**
+     * Creates and manages the setting "indicators" list.
+     */
+    fluid.defaults("gpii.qssWidget.mouse.indicators", {
+        gradeNames: "gpii.psp.repeater",
+
+        //
+        // Repeater stuff
+        //
+        dynamicContainerMarkup: {
+            container: "<div role='radio' class='%containerClass fl-qssStepperWidget-indicator' tabindex='-1'></div>",
+            containerClassPrefix: "flc-qssStepperWidget-indicator"
+        },
+        handlerType: "gpii.qssWidget.mouse.indicator.presenter",
+        markup: null,
+
+        //
+        // Custom
+        //
+        model: {
+            setting: {}
+        },
+        // Build the repeater items
+        modelRelay: {
+            items: {
+                target: "items",
+                singleTransform: {
+                    type: "fluid.transforms.free",
+                    func: "gpii.qssWidget.mouse.getIndicatorsList",
+                    args: [
+                        "{gpii.qssWidget.mouse.indicators}.model.setting"
+                    ]
+                }
+            }
+        },
+
+        events: {
+            onIndicatorClicked: null
+        },
+
+        listeners: {
+            "onIndicatorClicked.updateValue": {
+                changePath: "setting.value",
+                value: "{arguments}.0",
+                source: "fromWidget"
+            }
+        }
+    });
+
+    /**
+     * Generates the different indicators' data based on a setting .
+     * indicators are generated using the setting's `min`, `max` and `divisibleBy` properties.
+     * In case either of those is missing, no indicators will be generated.
+     * Note that items will be recomputed every time the setting changes but only items that
+     * need to be re-rendered will do so (changeApplier merges the values).
+     * @param {Object} setting - The setting for which indicators must be created
+     * @return {Object[]} - The list of data for each indicator element. In case no indicators
+     * can be generated an empty array is returned
+     */
+    gpii.qssWidget.mouse.getIndicatorsList = function (setting) {
```

Comment:
As well as being mostly identical with the baseStepper version, this impl contains a dangerous floating-point bounded loop. If we definitely need to handle decimal values, the only safe approach is to multiply up the divisibleBy step by 100 or 1000 etc. so that it becomes an integer, otherwise we run the risk of missing the upper loop bound due to floating point error. Whilst fixing both these impls, also try to unify them so that the special case for value 0 is factored out somehow - why is it that 0 is treated specially by the mouse settings?